lib/os.js

```diff
@@ -82,31 +82,29 @@ const getNetworkInterfacesDepMsg =
   'os.getNetworkInterfaces is deprecated. Use os.networkInterfaces instead.';
 
 const avgValues = new Float64Array(3);
-const cpuValues = new Float64Array(6 * pushValToArrayMax);
 
 function loadavg() {
   getLoadAvg(avgValues);
   return [avgValues[0], avgValues[1], avgValues[2]];
 }
 
-function addCPUInfo() {
-  for (var i = 0, c = 0; i < arguments.length; ++i, c += 6) {
-    this[this.length] = {
-      model: arguments[i],
-      speed: cpuValues[c],
+function cpus() {
+  const data = getCPUs();
+  const result = [];
```

Comment:
You could do the same trick (preallocate `const result = new Array(data.length / 7);`)

Reply:
Question: why not construct `result` in C++? For readability?

Reply:
@refack `Object::Set` is why `pushValuesToArray` exist in the first place: https://github.com/nodejs/node/pull/3780 I believe it's still pretty slow considering the hoops it needs to jump through

Reply:
> You could do the same trick (preallocate const result = new Array(data.length / 7);)

Doing this in v8 gives you a JSArray with HOLEY_ELEMENTS instead of PACKED_ELEMENTS, which is slower

Reply:
> `Object::Set` being slow

Ack. Thanks.

[academic discussion]
If the code in JS is more readable, that's also a plus. The only problem is these two functions (`cpus()` and `getCPUs()`) are tightly coupled but have a very loose API contract...

Worth thinking about - since this is probably a cold path I'd prefer we optimize for maintainability and readability, not speed.

Reply:
@refack TBH I find it much more readable to construct objects in JS instead of writing JS in C++ APIs..

Reply:
>  The only problem is these two functions (cpus() and getCPUs()) are tightly coupled but have a very loose API contract...

This pretty much describes everything in `internalBinding` and `process.binding`?